### PR TITLE
Fixes for M4 booting

### PR DESCRIPTION
**Pull request by mike-scott**

### CODE REVIEW

**Comment by mike-scott** [comment]
Testing this with the last patch to see if it "fixes" OP-TEE overlay generation

**Comment by mike-scott** [comment]
Ok, this is probably good for now.

File: drivers/fpga/imx_m4.c [hunk 1]
```diff
@@ -397,6 +391,13 @@ int fpga_loadbitstream(int d, char *bitstream, size_t size, bitstream_type t)
 		return ret;
 	}
 #endif
+
+	/* check if the M4 is already running */
```
Comment (ldts) [inline drivers/fpga/imx_m4.c:395]: looks good to me - I guess this value is always 0 in single boot mode? also across warm resets?

File: arch/arm/dts/imx7ulp.dtsi [hunk 2]
```diff
@@ -62,9 +62,9 @@
 			linux,cma-default;
 		};
 
-		rpmsg_reserved: rpmsg@9FFF0000 {
+		rpmsg_reserved: rpmsg@9DF00000 {
 			no-map;
-			reg = <0x9FF00000 0x100000>;
```
Comment (ldts) [inline arch/arm/dts/imx7ulp.dtsi:67]: looks good. just below TZDRAM so no collisions.

File: drivers/fpga/imx_m4.c (hunk 1 above)
Comment (mike-scott) [reply drivers/fpga/imx_m4.c:395]: It's 0 actually on a warm reset.. which means we would perform a hot update of the QSPI if the SHA was different ... but now that we're known running in TCM.. I'm not sure it matters?

**Pull request merged**